quantum/quantum.c

```diff
@@ -982,6 +982,10 @@ void matrix_scan_quantum() {
     matrix_scan_combo();
 #endif
 
+#if defined(LEADER_ENABLE) && defined(LEADER_ON_KEY_PROCESSING)
```

Comment:
I think we could just use:
```suggestion
#if defined(LEADER_ENABLE)
```

Reply:
I think this would break existing code because calling `matrix_scan_leader()` when on key processing is not enabled would finish the leader mode before the user gets the chance to catch it in his `matrix_scan_user()` with the usual `LEADER_DICTIONARY()` method.

However, we could move the check to into `matrix_scan_leader()` to keep the matrix function cleaner.